src/Hosting/Hosting/src/Host.cs

```diff
@@ -70,9 +70,11 @@ public static IHostBuilder CreateDefaultBuilder(string[] args)
             builder.ConfigureAppConfiguration((hostingContext, config) =>
             {
                 var env = hostingContext.HostingEnvironment;
+                
+                var reloadOnChange = hostingContext.Configuration.GetValue("HOSTBUILDER_CONFIG_RELOAD", true);
```

Comment:
The host uses top level keys like `contentRoot`.
https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/generic-host?view=aspnetcore-3.1#contentrootpath
`configReload` should be adequate here.

Reply:
Is there some prefix I can add to that to disambiguate? I was trying to follow the `{ASPNETCORE/DOTNET}_FEATURE_DESCRIPTION` nomenclature of other aspnetcore and dotnet flags sort of like [DOTNET_USE_POLLING_FILE_WATCHER](https://github.com/dotnet/aspnetcore/blob/6cc9f6f130af4ed0e7f321b144265cfbcec0ceee/src/Tools/dotnet-watch/src/CommandLineOptions.cs#L26)

This would mean that `DOTNET_CONFIGRELOAD` specified in an environment variable would trip the behavior added in this PR, and that to me sounds like it's entirely disabling config reloading, which is not quite the case, as one's own calls to `AddJsonFile` are left alone.

Reply:
We can at least use the hierarchical config pattern.
Also, config keys are not case sensitive. Changed for readability.
```suggestion
            var reloadFlagConfig = new Dictionary<string, string>() {{ "hostbuilder:configreload", "false" }};
```

Reply:
Sounds good, this has been updated in my most recent commit https://github.com/dotnet/extensions/pull/2940/commits/de16b88781a0a8171a6268a43f8b1211679bdc8b